WDL/runtime/cache.py

```diff
@@ -89,7 +98,10 @@ def get(
             return None
         contents = json.loads(contents)
         self._logger.notice(_("call cache hit", cache_path=file_path))  # pyre-fixme
-        return values_from_json(contents, output_types)  # pyre-fixme
+        cache = values_from_json(contents, output_types)  # pyre-fixme
+        files = Value.rewrite_env_files(cache, lambda file: file)
+        if file_coherence_checker.check_output_files(file_path, files):
```

Comment:
We need to check coherence of the files referenced in the task inputs as well as the outputs. What I had to walk back before was, we don't need to *store* the inputs in the cache file, because if we've made it this far then we must already have them in hand (we found the identical hash key).

This `get` method is only receiving the cache key so we'll probably have to refactor some stuff to give it the full inputs environment too.